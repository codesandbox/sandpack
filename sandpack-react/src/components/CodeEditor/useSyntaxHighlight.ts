import type { HighlightStyle } from "@codemirror/highlight";
import { highlightTree } from "@codemirror/highlight";
import type { LanguageSupport } from "@codemirror/language";
import { createElement } from "react";

export const useSyntaxHighlight = ({
  langSupport,
  highlightTheme,
  code = "",
}: {
  langSupport: LanguageSupport;
  highlightTheme: HighlightStyle;
  code?: string;
}): React.ReactNode[] => {
  const tree = langSupport.language.parser.parse(code);

  let offSet = 0;
  const codeElementsRender = [] as React.ReactNode[];

  const addElement = (to: number, className: string): void => {
    if (to > offSet) {
      const children = code.slice(offSet, to);

      codeElementsRender.push(
        className
          ? createElement("span", {
              children,
              className,
              key: `${to}${offSet}`,
            })
          : children
      );

      offSet = to;
    }
  };

  highlightTree(tree, highlightTheme.match, (from, to, className) => {
    addElement(from, "");
    addElement(to, className);
  });

  /**
   * The language parse doesn't look consistent.
   * The final syntax highlight used by CodeMirror
   * includes an end empty line, and the parse here doesn't,
   * so let's add it manually.
   */
  if (offSet < code.length) {
    codeElementsRender.push("\n\n");
  }

  return codeElementsRender;
};
