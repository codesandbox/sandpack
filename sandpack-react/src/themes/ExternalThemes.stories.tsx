import * as allThemes from "@codesandbox/sandpack-themes";
import { storiesOf } from "@storybook/react";
import React from "react";

import { Sandpack } from "../";

const stories = storiesOf("presets/Themes (external)", module);

Object.entries(allThemes).forEach(([themeName, value]) =>
  stories.add(themeName, () => (
    <Sandpack
      options={{ showInlineErrors: true }}
      template="react"
      theme={value}
    />
  ))
);
