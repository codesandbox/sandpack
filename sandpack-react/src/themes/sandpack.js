const customTheme = {
  palette: {
    activeText: "#1f2933",
    defaultText: "#757678",
    inactiveText: "#e4e7eb",
    activeBackground: "#e4e7eb",
    defaultBackground: "#f8f9fb",
    inputBackground: "#ffffff",
    accent: "#64D2FF",
    errorBackground: "#ffcdca",
    errorForeground: "#811e18",
  },
  syntax: {
    plain: "#151515",
    comment: { color: "#999", fontStyle: "italic" },
    keyword: "#0971F1",
    tag: "#0971F1",
    punctuation: "#151515",
    definition: "#151515",
    property: "#151515",
    static: "#FF453A",
    string: "#BF5AF2",
  },
  typography: {
    bodyFont:
      '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"',
    monoFont:
      '"Fira Mono", "DejaVu Sans Mono", Menlo, Consolas, "Liberation Mono", Monaco, "Lucida Console", monospace',
    fontSize: "14px",
    lineHeight: "1.4",
  },
};

exports.module = customTheme