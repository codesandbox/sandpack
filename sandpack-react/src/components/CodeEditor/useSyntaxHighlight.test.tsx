import { python } from "@codemirror/lang-python";
import { LanguageSupport } from "@codemirror/language";
import { shell } from "@codemirror/legacy-modes/mode/shell";
import { StreamLanguage } from "@codemirror/stream-parser";
import React from "react";
import { create } from "react-test-renderer";

import { defaultLight } from "../../themes";

import * as mocks from "./languages-mocks";
import { useSyntaxHighlight } from "./useSyntaxHighlight";
import {
  getCodeMirrorLanguage,
  getSyntaxHighlight,
  getLanguageFromFile,
} from "./utils";

const highlightTheme = getSyntaxHighlight(defaultLight);
const additionalLanguages = [
  {
    name: "shell",
    extensions: ["sh"],
    language: new LanguageSupport(StreamLanguage.define(shell)),
  },
  {
    name: "python",
    extensions: ["py"],
    language: python(),
  },
];

describe(useSyntaxHighlight, () => {
  Object.entries(mocks).forEach(([fileType, code]) => {
    it(`renders a ${fileType} block`, () => {
      const languageExtension = getLanguageFromFile(
        "",
        fileType,
        additionalLanguages
      );
      const langSupport = getCodeMirrorLanguage(
        languageExtension,
        additionalLanguages
      );

      const reactElements = useSyntaxHighlight({
        code,
        highlightTheme,
        langSupport,
      });

      expect(create(<>{reactElements}</>).toJSON()).toMatchSnapshot();
    });
  });
});
