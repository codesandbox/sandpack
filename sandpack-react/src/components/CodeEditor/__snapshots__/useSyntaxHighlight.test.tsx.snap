// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useSyntaxHighlight renders a css block 1`] = `
Array [
  <span
    className="sp-syntax-tag"
  >
    body
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  "
  ",
  <span
    className="sp-syntax-property"
  >
    background-color
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  " ",
  <span
    className="sp-syntax-static"
  >
    lightblue
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
  "

",
  <span
    className="sp-syntax-tag"
  >
    h1
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  "
  ",
  <span
    className="sp-syntax-property"
  >
    color
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  " ",
  <span
    className="sp-syntax-static"
  >
    white
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
  ",
  <span
    className="sp-syntax-property"
  >
    text-align
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  " ",
  <span
    className="sp-syntax-static"
  >
    center
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
  "

",
  <span
    className="sp-syntax-tag"
  >
    p
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  "
  ",
  <span
    className="sp-syntax-property"
  >
    font-family
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  " ",
  <span
    className="sp-syntax-static"
  >
    verdana
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
  ",
  <span
    className="sp-syntax-property"
  >
    font-size
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  " ",
  <span
    className="sp-syntax-static"
  >
    20
  </span>,
  <span
    className="sp-syntax-keyword"
  >
    px
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
]
`;

exports[`useSyntaxHighlight renders a html block 1`] = `
Array [
  "
<!DOCTYPE html>
",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    html
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    lang
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "en"
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
  ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    head
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
    ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    meta
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    charset
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "UTF-8"
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
    ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    meta
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    name
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "viewport"
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    content
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "width=device-width, initial-scale=1.0"
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
    ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    meta
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    http-equiv
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "X-UA-Compatible"
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    content
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "ie=edge"
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
    ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    title
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "HTML 5 Boilerplate",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    title
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
    ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    link
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    rel
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "stylesheet"
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    href
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "style.css"
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
  ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    head
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
  ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    body
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
  ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    script
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    src
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "index.js"
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    script
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
  ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    body
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    html
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "

",
]
`;

exports[`useSyntaxHighlight renders a js block 1`] = `
Array [
  <span
    className="sp-syntax-keyword"
  >
    function
  </span>,
  " ",
  <span
    className="sp-syntax-definition"
  >
    foo
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    (
  </span>,
  <span
    className="sp-syntax-plain"
  >
    params
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    )
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  "
  ",
  <span
    className="sp-syntax-keyword"
  >
    return
  </span>,
  " ",
  <span
    className="sp-syntax-plain"
  >
    params
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
  "

",
  <span
    className="sp-syntax-keyword"
  >
    const
  </span>,
  " ",
  <span
    className="sp-syntax-plain"
  >
    baz
  </span>,
  " = ",
  <span
    className="sp-syntax-punctuation"
  >
    (
  </span>,
  <span
    className="sp-syntax-plain"
  >
    foo
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    )
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    =&gt;
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  "
  ",
  <span
    className="sp-syntax-keyword"
  >
    return
  </span>,
  " ",
  <span
    className="sp-syntax-plain"
  >
    foo
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
  "

",
  <span
    className="sp-syntax-keyword"
  >
    const
  </span>,
  " ",
  <span
    className="sp-syntax-plain"
  >
    array
  </span>,
  " = ",
  <span
    className="sp-syntax-punctuation"
  >
    [
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ]
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
",
  <span
    className="sp-syntax-keyword"
  >
    const
  </span>,
  " ",
  <span
    className="sp-syntax-plain"
  >
    object
  </span>,
  " = ",
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
",
  <span
    className="sp-syntax-keyword"
  >
    const
  </span>,
  " ",
  <span
    className="sp-syntax-plain"
  >
    regex
  </span>,
  " = ",
  <span
    className="sp-syntax-keyword"
  >
    new
  </span>,
  " ",
  <span
    className="sp-syntax-plain"
  >
    Regex
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    (
  </span>,
  <span
    className="sp-syntax-comment"
  >
    //);
  </span>,
  "

",
  <span
    className="sp-syntax-keyword"
  >
    const
  </span>,
  " ",
  <span
    className="sp-syntax-plain"
  >
    expr
  </span>,
  " = ",
  <span
    className="sp-syntax-string"
  >
    'Papayas'
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
",
  <span
    className="sp-syntax-keyword"
  >
    switch
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    (
  </span>,
  <span
    className="sp-syntax-plain"
  >
    expr
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    )
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  "
  ",
  <span
    className="sp-syntax-keyword"
  >
    case
  </span>,
  " ",
  <span
    className="sp-syntax-string"
  >
    'Oranges'
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  "
    ",
  <span
    className="sp-syntax-plain"
  >
    console
  </span>,
  ".",
  <span
    className="sp-syntax-property"
  >
    log
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    (
  </span>,
  <span
    className="sp-syntax-string"
  >
    'Oranges are $0.59 a pound.'
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    )
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
    ",
  <span
    className="sp-syntax-keyword"
  >
    break
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
  ",
  <span
    className="sp-syntax-keyword"
  >
    case
  </span>,
  " ",
  <span
    className="sp-syntax-string"
  >
    'Mangoes'
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  "
  ",
  <span
    className="sp-syntax-keyword"
  >
    case
  </span>,
  " ",
  <span
    className="sp-syntax-string"
  >
    'Papayas'
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  "
    ",
  <span
    className="sp-syntax-plain"
  >
    console
  </span>,
  ".",
  <span
    className="sp-syntax-property"
  >
    log
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    (
  </span>,
  <span
    className="sp-syntax-string"
  >
    'Mangoes and papayas are $2.79 a pound.'
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    )
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
    ",
  <span
    className="sp-syntax-comment"
  >
    // expected output: "Mangoes and papayas are $2.79 a pound."
  </span>,
  "
    ",
  <span
    className="sp-syntax-keyword"
  >
    break
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
  ",
  <span
    className="sp-syntax-keyword"
  >
    default
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  "
    ",
  <span
    className="sp-syntax-plain"
  >
    console
  </span>,
  ".",
  <span
    className="sp-syntax-property"
  >
    log
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    (
  </span>,
  <span
    className="sp-syntax-string"
  >
    \`Sorry, we are out of 
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    \${
  </span>,
  <span
    className="sp-syntax-plain"
  >
    expr
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
  <span
    className="sp-syntax-string"
  >
    .\`
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    )
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
]
`;

exports[`useSyntaxHighlight renders a jsx block 1`] = `
Array [
  <span
    className="sp-syntax-keyword"
  >
    function
  </span>,
  " ",
  <span
    className="sp-syntax-definition"
  >
    Greeting
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    (
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    name
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    )
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  "
  ",
  <span
    className="sp-syntax-keyword"
  >
    return
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    h1
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "Hello, ",
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  <span
    className="sp-syntax-plain"
  >
    name
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
  "!",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    h1
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
  "

",
  <span
    className="sp-syntax-keyword"
  >
    export
  </span>,
  " ",
  <span
    className="sp-syntax-keyword"
  >
    default
  </span>,
  " ",
  <span
    className="sp-syntax-keyword"
  >
    function
  </span>,
  " ",
  <span
    className="sp-syntax-definition"
  >
    App
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    (
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    )
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  "
  ",
  <span
    className="sp-syntax-keyword"
  >
    return
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    (
  </span>,
  "
    ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    div
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
      ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    Greeting
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    name
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "Divyesh"
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    /&gt;
  </span>,
  "
      ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    Greeting
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    name
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "Sarah"
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    /&gt;
  </span>,
  "
      ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    Greeting
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    name
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "Taylor"
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    /&gt;
  </span>,
  "
    ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    div
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
  ",
  <span
    className="sp-syntax-punctuation"
  >
    )
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
]
`;

exports[`useSyntaxHighlight renders a less block 1`] = `
Array [
  <span
    className="sp-syntax-keyword"
  >
    @width
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  " 10",
  <span
    className="sp-syntax-keyword"
  >
    px
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
",
  <span
    className="sp-syntax-keyword"
  >
    @height
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  " ",
  <span
    className="sp-syntax-keyword"
  >
    @width
  </span>,
  " + 10",
  <span
    className="sp-syntax-tag"
  >
    px
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "

#header ",
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  "
  ",
  <span
    className="sp-syntax-property"
  >
    width
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  " @",
  <span
    className="sp-syntax-static"
  >
    width
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
  ",
  <span
    className="sp-syntax-property"
  >
    height
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  " @",
  <span
    className="sp-syntax-static"
  >
    height
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
]
`;

exports[`useSyntaxHighlight renders a python block 1`] = `
Array [
  <span
    className="sp-syntax-keyword"
  >
    import
  </span>,
  " ",
  <span
    className="sp-syntax-plain"
  >
    os
  </span>,
  "

",
  <span
    className="sp-syntax-plain"
  >
    api_token
  </span>,
  " = ",
  <span
    className="sp-syntax-plain"
  >
    os
  </span>,
  ".",
  <span
    className="sp-syntax-property"
  >
    environ
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    [
  </span>,
  <span
    className="sp-syntax-string"
  >
    'API_TOKEN'
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ]
  </span>,
  "

",
  <span
    className="sp-syntax-comment"
  >
    # Prints the api_token variable
  </span>,
  "
",
  <span
    className="sp-syntax-definition"
  >
    print
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    (
  </span>,
  <span
    className="sp-syntax-string"
  >
    "API token:"
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ,
  </span>,
  " ",
  <span
    className="sp-syntax-plain"
  >
    api_token
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    )
  </span>,
]
`;

exports[`useSyntaxHighlight renders a shell block 1`] = `
Array [
  "#!/bin/sh

",
  <span
    className="sp-syntax-plain"
  >
    EXAMPLE
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "drawn joyed"
  </span>,
  "

",
  <span
    className="sp-syntax-comment"
  >
    # Prints the EXAMPLE variable
  </span>,
  "
",
  <span
    className="sp-syntax-keyword"
  >
    function
  </span>,
  " show-example() {
  ",
  <span
    className="sp-syntax-plain"
  >
    echo
  </span>,
  " ",
  <span
    className="sp-syntax-plain"
  >
    $EXAMPLE
  </span>,
  "

",
]
`;

exports[`useSyntaxHighlight renders a vue block 1`] = `
Array [
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    template
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    div
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    class
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "column is-12"
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
    ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    label
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    class
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "label"
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    for
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "email"
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "Email",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    label
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
    ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    p
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    :class
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "{ 'control': true }"
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
        ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    input
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    v-validate
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "'required|email'"
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    :class
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "{'input': true, 'is-danger': errors.has('email') }"
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    name
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "email"
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    type
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "text"
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    placeholder
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "Email"
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
        ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    span
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    v-show
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "errors.has('email')"
  </span>,
  " ",
  <span
    className="sp-syntax-property"
  >
    class
  </span>,
  "=",
  <span
    className="sp-syntax-string"
  >
    "help is-danger"
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "{{ errors.first('email') }}",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    span
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
    ",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    p
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    div
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    template
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "

",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;
  </span>,
  <span
    className="sp-syntax-tag"
  >
    script
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
  "
  ",
  <span
    className="sp-syntax-keyword"
  >
    export
  </span>,
  " ",
  <span
    className="sp-syntax-keyword"
  >
    default
  </span>,
  " ",
  <span
    className="sp-syntax-punctuation"
  >
    {
  </span>,
  "
  ",
  <span
    className="sp-syntax-property"
  >
    name
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    :
  </span>,
  " ",
  <span
    className="sp-syntax-string"
  >
    'basic-example'
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    }
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    ;
  </span>,
  "
",
  <span
    className="sp-syntax-punctuation"
  >
    &lt;/
  </span>,
  <span
    className="sp-syntax-tag"
  >
    script
  </span>,
  <span
    className="sp-syntax-punctuation"
  >
    &gt;
  </span>,
]
`;
